plutus-core/plutus-core/src/PlutusCore/Bitwise.hs

```diff
@@ -567,28 +567,28 @@ writeBits bs ixs bits = case unsafeDupablePerformIO . try $ go of
     go = BS.useAsCString bs $ \srcPtr ->
           BSI.create len $
             \dstPtr ->
-              let go2 (i:is) (v:vs) = setAtIx dstPtr i v *> go2 is vs
-                  go2 _ _           = pure ()
+              let go2 (i:is) = setOrClearAtIx dstPtr i *> go2 is
+                  go2 _      = pure ()
               in do
                 copyBytes dstPtr (castPtr srcPtr) len
-                go2 ixs bits
+                go2 ixs
     len :: Int
     len = BS.length bs
     bitLen :: Integer
     bitLen = fromIntegral len * 8
-    setAtIx :: Ptr Word8 -> Integer -> Bool -> IO ()
-    setAtIx ptr i b
+    setOrClearAtIx :: Ptr Word8 -> Integer -> IO ()
+    setOrClearAtIx ptr i
       | i < 0 = throw $ WriteBitsException i
       | i >= bitLen = throw $ WriteBitsException i
```

Comment:
While you're here, please replace these two `throw` calls with `throwIO` ones. The Haddock of the latter:

```
-- | A variant of 'throw' that can only be used within the 'IO' monad.
--
-- Although 'throwIO' has a type that is an instance of the type of 'throw', the
-- two functions are subtly different:
--
-- > throw e   `seq` ()  ===> throw e
-- > throwIO e `seq` ()  ===> ()
--
-- The first example will cause the exception @e@ to be raised,
-- whereas the second one won\'t.  In fact, 'throwIO' will only cause
-- an exception to be raised when it is used within the 'IO' monad.
--
-- The 'throwIO' variant should be used in preference to 'throw' to
-- raise an exception within the 'IO' monad because it guarantees
-- ordering with respect to other operations, whereas 'throw'
-- does not. We say that 'throwIO' throws *precise* exceptions and
-- 'throw', 'error', etc. all throw *imprecise* exceptions.
-- For example
--
-- > throw e + error "boom" ===> error "boom"
-- > throw e + error "boom" ===> throw e
--
-- are both valid reductions and the compiler may pick any (loop, even), whereas
--
-- > throwIO e >> error "boom" ===> throwIO e
--
-- will always throw @e@ when executed.
--
-- See also the
-- [GHC wiki page on precise exceptions](https://gitlab.haskell.org/ghc/ghc/-/wikis/exceptions/precise-exceptions)
-- for a more technical introduction to how GHC optimises around precise vs.
-- imprecise exceptions.
```

Using `throw` here is asking for trouble.

Reply:
> While you're here, please replace these two throw calls with throwIO ones.

OK, I done that.  Thanks for pointing that out.